webpack.config.js

```diff
@@ -21,46 +20,204 @@ module.exports = {
         "theme-light-custom": "./node_modules/matrix-react-sdk/res/themes/light-custom/css/light-custom.scss",
         "theme-dark-custom": "./node_modules/matrix-react-sdk/res/themes/dark-custom/css/dark-custom.scss",
     },
+
+    optimization: {
+        // Put all of our CSS into one useful place - this is needed for MiniCssExtractPlugin.
+        // Previously we used a different extraction plugin that did this magic for us, but
+        // now we need to consider that the CSS needs to be bundled up together.
+        splitChunks: {
+            cacheGroups: {
+                styles: {
+                    name: 'styles',
+                    test: /\.css$/,
+                    enforce: true,
+                    // Do not add `chunks: 'all'` here because you'll break the app entry point.
+                },
+            },
+        },
+
+        // Minification is normally enabled by default for webpack in production mode, but
+        // we use a CSS optimizer too and need to manage it ourselves.
+        minimize: argv.mode === 'production',
+        minimizer: argv.mode === 'production' ? [new TerserPlugin({}), new OptimizeCSSAssetsPlugin({})] : [],
```

Comment:
Great that we have minified CSS now, but this seems like another thing that didn't really need to be part of this epic journey.

Reply:
sorry :(

I got carried away in places.

Reply:
Well, probably some advantage in tackling it whilst source map knowledge is swapped in.